parsl/app/futures.py

```diff
@@ -31,15 +34,26 @@ def parent_callback(self, parent_fu):
         Returns:
             - None
         """
-
         e = parent_fu._exception
         if e:
             self.set_exception(e)
         else:
             self.set_result(self.file_obj)
+            if self.file_obj.scheme == 'file' and os.path.isfile(self.file_obj.filepath):
+                if not self.file_obj.timestamp:
+                    self.file_obj.timestamp = datetime.fromtimestamp(stat(self.file_obj.filepath).st_ctime, tz=timezone.utc)
+                if not self.file_obj.size:
+                    self.file_obj.size = stat(self.file_obj.filepath).st_size
+                if not self.file_obj.md5sum:
+                    self.file_obj.md5sum = md5(open(self.file_obj, 'rb').read()).hexdigest()
+            else:
+                self.file_obj.timestamp = datetime.now(tz=timezone.utc)
```

Comment:
I feel like this is a bit suspicious - in the `else` case you don't know the timestamp and maybe you should admit to that rather than inventing one?

Reply:
ok, fixed